TypedSyntax/src/show.jl

```diff
@@ -43,19 +44,44 @@ function Base.printstyled(io::IO, rootnode::MaybeTypedSyntaxNode;
         @assert length(children(rootnode)) == 2
         sig, body = children(rootnode)
         type_annotate, pre, pre2, post = type_annotation_mode(sig, rt; type_annotations, hide_type_stable)
-        position = show_src_expr(io, sig, position, pre, pre2; type_annotations, iswarn, hide_type_stable, nd)
-        type_annotate && show_annotation(io, rt, post; iswarn)
+        position = show_src_expr(io, sig, position, pre, pre2, type_hints, warn_diagnostics; type_annotations, iswarn, hide_type_stable, nd)
+        type_annotate && show_annotation(io, rt, post, rootnode.source, position, type_hints, warn_diagnostics; iswarn)
         rootnode = body
     end
-    position = show_src_expr(io, rootnode, position, "", ""; type_annotations, iswarn, hide_type_stable, nd)
+    position = show_src_expr(io, rootnode, position, "", "", type_hints, warn_diagnostics; type_annotations, iswarn, hide_type_stable, nd)
     catchup(io, rootnode, position, nd, idxend+1)   # finish the node
     return nothing
 end
-Base.printstyled(rootnode::MaybeTypedSyntaxNode; kwargs...) = printstyled(stdout, rootnode; kwargs...)
+function Base.printstyled(io::IO, rootnode::MaybeTypedSyntaxNode, 
+                          type_hints = Dict{String, Vector{InlayHint}}(), 
+                          warn_diagnostics = WarnUnstable[];
```

Comment:
What's the reason for making some of these positional and the others keywords?

I think my main question would be: could we clean this up by adding these to the `io` via an `IOContext`? There appear to be good arguments for that, including that they are both outputs rather than inputs (it's a little weird to have the argument role go `(Output, Input, Output, Output)`), and `Base.printstyled` docstring specifies that any additional arguments get printed to `io` (and not *printed to*). It might also let this be less intrusive of a change.